src/test/resources/hudson/plugins/copyartifact/maven-job/pom.xml

```diff
@@ -0,0 +1,28 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>org.jvnet.hudson.main.test.multimod</groupId>
+  <artifactId>multimod-top</artifactId>
+  <packaging>pom</packaging>
+  <version>1.0-SNAPSHOT</version>
+  <name>multimod-top</name>
+  <description>Multimodule project with three modules, none dependent on each other.</description>
+  <url>http://maven.apache.org</url>
+  <modules>
+    <module>moduleA</module>
+    <module>moduleB</module>
+    <module>moduleC</module>
+  </modules>
+  <repositories>
+    <repository>
+      <id>central</id>
+      <url>https://repo.maven.apache.org/maven2</url>
+    </repository>
+  </repositories>
+  <pluginRepositories>
+    <pluginRepository>
+      <id>central</id>
+      <url>https://repo.maven.apache.org/maven2</url>
+    </pluginRepository>
+  </pluginRepositories>
```

Comment:
I added this to all pom files.
Adding this only to the root pom doesn't resolve the problem. It might caused for submodules doesn't inherit the parent module explicitly. (I don't know about detailed behavior of maven)
Adding this only to moduleA might resolve the problem as the artifacts will be downloaded for moduleA, and cached and reused for moduleB, moduleC. But that's just a workaround and not a good solution.